configs/BUILD

```diff
@@ -60,10 +60,11 @@ genrule(
         "//docs:configs",
         "//docs:proto_examples",
         "//test/config/integration/certs",
+        "//examples/secrets",
     ],
     outs = ["example_configs.tar"],
     cmd = (
-        "$(location configgen.sh) $(location configgen) example_configs.tar $(@D) " +
+        "$(location configgen.sh) $(location configgen) $(locations //examples/secrets) example_configs.tar $(@D) " +
```

Comment:
this should be bundled into `example_configs.tar` then there would be no need to add this here

Reply:
oh wait - checked the script and realized that its using these args - i still think its not quite right

`locations` can produce multiple paths - whereas this assumes a single path - i think this needs to be placed after the `OUT_DIR` and treated as args

Reply:
This(secrets in .yaml files) will be bundled into `example_configs.tar` only. If you look at changes which I added in `configgen.sh`, This argument is used to figure out secrets dir path and then all yamls from the source `secrets` directory are copied into a new created `secrets` directory which is bundled into `example_configs.tar`. 
core of the issue is that existing logic in `configgen.sh` treats all yaml files as configs to be tests i.e files with which server is bootstrapped.

Reply:
iirc it also has the lua and wasm artefacts - i think this is more like those

either way this needs to be fixed as the `locations` cant be assumed to be a single file

Reply:
>locations can produce multiple paths - whereas this assumes a single path

right. I could not find a way to pass dir path as argument. That why in [examples/secrets/BUILD](https://github.com/envoyproxy/envoy/pull/33702/files#diff-15feed8cd0e27e7262826d97b261d555e63f44480bca88ae07a0adc19deac24bR13) not using `*` rather a single file name, so that it always return a single value

Reply:
you could use a single file rather than filegroup - but i think the latter (ie how you have done it) is much preferable

the mangling code is 

```bash
 20 for FILE in "$@"; do
 21   case "$FILE" in
 22   *.pem|*.der)
 23     cp "$FILE" "$OUT_DIR/certs"
 24     ;;
 25   *.lua|*.wasm|*.so)
 26     cp "$FILE" "$OUT_DIR/lib"
 27     ;;
 28   *.pb)
 29     cp "$FILE" "$OUT_DIR/protos"
 30     ;;
 31   *)
```

how about naming the file `*.credentials.yaml` which could be filtered in the script to place it in the lib dir or similar